### PR TITLE
Protected route

### PR DESCRIPTION
**Pull request by NancyRike**
* work on protected routes using react-router-dom, to send users that are  not logged in back to the sign-in page

### CODE REVIEW

**Review by Dverybest — COMMENTED** [review]
Remove this, if it is not used in the code

File: src/App.tsx [hunk 1]
```diff
@@ -8,7 +8,8 @@ import Home from "./pages/dashboard/Home";
 import Orders from "./pages/dashboard/Order";
 import Menu from "./pages/dashboard/menu/Menu";
 import Users from "./pages/dashboard/Users";
-import { MainRoute } from "./components/mainRoute";
+import MainRoute from "./components/mainRoute";
+// import {ProtectedRoute} from "./components/protectedRoute";
```
Comment (Dverybest) [inline src/App.tsx:12]: remove

File: src/components/mainRoute.tsx [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
   name,
   component: Component,
   ...rest
 }: IMainRoute) => {
+  const location = useLocation();
   return (
     <Route
       {...rest}
       render={(routeProps) => (
+        isAuth ===true?(
```
Comment (Dverybest) [inline src/components/mainRoute.tsx:25]: isAuth is a boolean no need of the check

File: src/interface/actionInterface.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface payloadType {
```
Comment (Dverybest) [inline src/interface/actionInterface.tsx:1]: Remove

File: src/interface/reducerInterface.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export interface IAction {
```
Comment (Dverybest) [inline src/interface/reducerInterface.tsx:1]: Move to actionInterface file

File: src/reduxStore/actions/authAction.ts [hunk 5]
```diff
@@ -1,30 +1,27 @@
...
+
       requestProcessor({
         method:'POST',
         url:'/admin/login',
         payload
-      }).then(response=>{
-        console.log(response, "na sooooo");
-        
+      }).then(response=>{        
         /**
```
Comment (Dverybest) [inline src/reduxStore/actions/authAction.ts:20]: Clear Todos if implemented

File: src/reduxStore/store.ts [hunk 6]
```diff
@@ -23,6 +23,8 @@ import { IAction } from "./Reducers/authReducer";
     const storeData = (store:MiddlewareAPI) => (next:Dispatch<IAction>) => (action:IAction) => {
         const saveData =  store.getState();
         sessionStorage.setItem('itemKey', JSON.stringify({...saveData}));
+        console.log({saveData});
```
Comment (Dverybest) [inline src/reduxStore/store.ts:26]: Remove console

**Review by Dverybest — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (NancyRike) [reply src/App.tsx:12]: Done

File: src/components/mainRoute.tsx (hunk 2 above)
Comment (NancyRike) [reply src/components/mainRoute.tsx:25]: Resolved

File: src/interface/actionInterface.tsx (hunk 3 above)
Comment (NancyRike) [reply src/interface/actionInterface.tsx:1]: payloadType interface removed

**Review by Dverybest — APPROVED** [review]

**Pull request merged**